bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -451,6 +451,31 @@ impl Connection {
         Ok(self)
     }
 
+    /// Configure the connection to reduce potentially expensive calls to recv.
+    ///
+    /// If this setting is disabled, s2n-tls will call read twice for every TLS record,
+    /// which can be expensive but ensures that s2n-tls will always attempt to read the
+    /// exact number of bytes it requires. If this setting is enabled, s2n-tls will
+    /// instead reduce the number of calls to read by attempting to read as much data
+    /// as possible in each read call, storing the extra in the existing IO buffers.
+    /// This may cause it to request more data than will ever actually be available.
+    ///
+    /// There is no additional memory cost of enabling this setting. It reuses the
+    /// existing IO buffers.
+    ///
+    /// This setting is disabled by default. Depending on how your application detects
+    /// data available for reading, buffering reads may break your event loop.
```

Comment:
I'm not sure about copy/pasting the C comment into the Rust. There's no mechanism to make sure that the two comments stay in sync. I prefer just linking to the source of truth, which is the C comment.

Reply:
fair comment, I'll pushed a change for just the brief description and defer the reader to the C docs